### PR TITLE
Delete last message updates thread preview (signal ios#833)

### LINKED ISSUE #833: Deleting the last message does not update conversation list snippet
1. Open a conversation that has more than one message
1. Delete the last message from that conversation
1. Return back to conversation list
1. The message snippet for that conversation is empty

iPhone 4S
iOS 8.3
Signal 2.1.1

(edit: modified the steps to be more explicit)

### PR DESCRIPTION
**Pull request by michaelkirk**
This fixes https://github.com/WhisperSystems/Signal-iOS/issues/833

Note that I had to merge #2 locally before I was able to build this.

img_0314

And I had to merge https://github.com/WhisperSystems/Signal-iOS/pull/1102 in order to delete messages to test this.

### CODE REVIEW

File: src/Contacts/TSThread.m [hunk 1]
```diff
@@ -19,8 +19,10 @@ @interface TSThread ()
 @property (nonatomic, retain) NSDate *creationDate;
 @property (nonatomic, copy) NSDate *archivalDate;
 @property (nonatomic, retain) NSDate *lastMessageDate;
-@property (nonatomic, copy) NSString *latestMessageId;
 @property (nonatomic, copy) NSString *messageDraft;
+
+- (TSInteraction *) lastInteraction;
+
 @end
```
Comment (FredericJacobs) [inline src/Contacts/TSThread.m:26]: Why would you store another serialization of that interaction? I don't see why you would duplicate it in the database in the thread.

Comment (michaelkirk) [inline src/Contacts/TSThread.m:26]: I'm not storing another serialization of the interaction - take a peak at the implementation a few lines down.

Essentially, I'm just taking advantage of our existing YapDatabase view which _already_ keeps track of lastObjectInGroup, vs us redundantly (and incorrectly) reimplementing some of that logic via our "lastMessageId" field.

edit: The reason this fixes (https://github.com/WhisperSystems/Signal-iOS/issues/833) is that we don't have to worry about updating the "lastMessageId" when the last message is deleted. Yap's already tracking it for us.

Comment (FredericJacobs) [inline src/Contacts/TSThread.m:26]: My bad. Read the GitHub diff to quickly.

Comment (michaelkirk) [inline src/Contacts/TSThread.m:26]: No worries. Just happy to see you in the PRs. =)

**Pull request merged**